hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/services/AbfsLease.java

```diff
@@ -88,6 +88,12 @@ public AbfsLease(AbfsClient client, String path) throws AzureBlobFileSystemExcep
     acquireLease(retryPolicy, 0, 0);
 
     while (leaseID == null && exception == null) {
+      try {
+        future.get();
+      } catch (Exception e) {
+        LOG.debug("Got exception waiting for acquire lease future. Checking if lease ID or "
+            + "exception have been set", e);
+      }
```

Comment:
if this raises an exception, is there any way the while loop will exit?

Reply:
Yes, in [this section](https://github.com/apache/hadoop/pull/1925/files#diff-5777ea1bdc7124c9ed93d1dd8981435a849e8cbf38b929792a1fc8d067dd6e79R135-R142) it will retry if it is below the max retries and otherwise set the exception variable. So by max retries we should either have a lease ID or an exception set, and the while loop will exit. In the unit test, I [mocked two failures followed by a success](https://github.com/apache/hadoop/pull/1925/files#diff-c952dccc1904f90c10dc7229d69482ebaa2f585f57761d7156d5899ee4207d5fR319-R327) as well as persistent failure and verified it had the correct behavior.

Reply:
understood